clients/cobol-lsp-vscode-extension/src/services/copybook/downloader/DownloadUtil.ts

```diff
@@ -183,11 +213,45 @@ export class DownloadUtil {
     return action === UNLOCK_DOWNLOAD_QUEUE_MSG;
   }
 
+  /**
+   * Checks if the error returned by Zowe Explorer is caused
+   * by invalid credentials. Error with status code 401 is returned
+   * in that case.
+   */
   private static isInvalidCredentials(e: unknown) {
     return (
       hasMember(e, "mDetails") &&
       hasMember(e.mDetails, "errorCode") &&
       e.mDetails.errorCode === 401
     );
   }
+
+  /**
+   * Returns true if provided credentials are correct but user doesn't
+   * have permission to access selected dataset (ISRZ002)
+   * or uss directory (EDC5111I).
+   */
+  private static isPermissionError(e: unknown) {
+    return (
+      hasMember(e, "mDetails") &&
+      hasMember(e.mDetails, "errorCode") &&
+      e.mDetails.errorCode === 500 &&
```

Comment:
We should test this behavior with RACF system as well. The 500 (caused by the S913 abend) could be very much specific to TSS and ACF2.